src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/Formatting/RazorFormattingService.cs

```diff
@@ -285,7 +299,33 @@ private async Task<ImmutableArray<TextChange>> ApplyFormattedChangesAsync(
                 return [];
             }
 
-            return [collapsedEdit];
+            ImmutableArray<TextChange> collapsedEdits = [collapsedEdit];
+
+            if (validate)
+            {
+                foreach (var validationPass in _validationPasses)
+                {
+                    var isValid = await validationPass.IsValidAsync(context, collapsedEdits, cancellationToken).ConfigureAwait(false);
+                    if (!isValid)
+                    {
+                        return [];
+                    }
+                }
+            }
```

Comment:
Can this be moved up to before `if (collapseChanges)`, and operate on `razorChanges`, so it doesn't need to be duplicated? Or is it possible collapsing could cause an issue?

Reply:
I don't think it will, and there's no bug known where it does, so it can probably be moved